### PR TITLE
New P97: Embeddable in the real line

### PR DESCRIPTION
**Pull request by prabau**
Add new property P97: Embeddable in $\mathbb R$

New theorems to accompany this:
- (T231) embeddable in R ==> embeddable in Euclidean space
- (T232) totally path disconnected + embeddable in R ==> zero dimensional
- (T233) connected + embeddable in R ==> locally path connected
- (updated T462) separable + metrizable + zero dimensional ==> embeddable in R
- (updated T463) LOTS + second countable ==> embeddable in R

These theorems allow to determine whether P97 holds or not for many spaces.  For all the remaining spaces known as "embeddable in Euclidean space" with P97 undetermined, added some explicit trait files as needed.

And more updates for S121 (Bernstein's connected set), S147 (Luzin set), S148 (Luzin set).

### CODE REVIEW

**Comment by ccaruvana** [comment]
Taking a look here and I have a quick question. This PR asserts P65 for S121, which is automatically deduced currently. I would assume that the change in asserted properties for S121 in this PR makes the automatic deduction of P65 impossible. Without taking an in depth look at the deduction tree (https://topology.pi-base.org/spaces/S000121/properties/P000065), is there another way to check why P65 isn't deduced for S121 with the content of this PR?

**Comment by ccaruvana** [comment]
Maybe this was mentioned in our last community call, but I've gotten a couple of 404s using "open link in new tab" for theorems and asserted properties in this PR while browsing. When I don't use the "open link in new tab" and just click directly, they seem to work fine and I can see no difference in the displayed URLs.

**Comment by StevenClontz** [comment]
So the 404 is triggering based upon the pages that exist in the `main` branch, when visited directly. As pointed out this is only when a page is loaded from scratch - internal routing works fine.

This is a wontfix for now as it only affects reviewers of non-main branches, as annoying as it is.

**Comment by ccaruvana** [comment]
I've checked over everything except for T463 in detail since it references a new mathse post which is involved. If no one else beats me to it, I'll think it about in more detail a bit later.

In the meantime, I believe what's written for T233, but would anyone think it beneficial to throw in some basic reference?

**Comment by prabau** [comment]
> Taking a look here and I have a quick question. This PR asserts P65 for S121, which is automatically deduced currently. I would assume that the change in asserted properties for S121 in this PR makes the automatic deduction of P65 impossible. Without taking an in depth look at the deduction tree (https://topology.pi-base.org/spaces/S000121/properties/P000065), is there another way to check why P65 isn't deduced for S121 with the content of this PR?

The previous justification for S121/P65 (https://topology.pi-base.org/spaces/S000121/properties/P000065) seemed extremely long, and depended on the asserted P129 (indiscrete) = false.  It seems simpler to assert P65 directly, as it's pretty immediate from the construction of the space, and make P129 deduced instead.

(Many spaces assert P129 directly, and don't need to.  I think it's a holdover of some old PR where the property was asserted for nearly all spaces.)

**Comment by prabau** [comment]
temporarily converted to Draft while I add a reference

**Comment by prabau** [comment]
> I believe what's written for T233, but would anyone think it beneficial to throw in some basic reference?

Not sure what to add, but feel free to modify T233 as you see fit.

**Comment by prabau** [comment]
Another general question (not specific to this PR, but affects it also). 
In the browser the reference names are not expanded for latex/mathjax.  See for example "mathbb R" in https://topology.pi-base.org/spaces/S000140/references.

Do you think it could be expanded at some point in the future?

**Comment by StevenClontz** [comment]
> Another general question (not specific to this PR, but affects it also). In the browser the reference names are not expanded for latex/mathjax. See for example "mathbb R" in https://topology.pi-base.org/spaces/S000140/references.
> 
> Do you think it could be expanded at some point in the future?

The issue is that when you write `{{mr:123456}}`, the component that renders it to a link is unaware of any reference data provided elsewhere in that text file or otherwise.

I'd prefer some kind of solution that allows us to only provide ids, and then some external API provides the title, e.g. https://api.zbmath.org/v1/

File: spaces/S000121/properties/P000065.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+space: S000121
+property: P000065
+value: true
+refs:
+- doi: 10.1007/978-1-4612-6290-9_6
+  name: Counterexamples in Topology
+---
+
+By construction.
```
Comment (StevenClontz) [inline spaces/S000121/properties/P000065.md:10]:
```suggestion
By construction: this space contains a distinct point chosen from $C_\alpha$ for each $\alpha<\mathfrak c$.
```

**Comment by prabau** [comment]
> The issue is that when you write `{{mr:123456}}`, the component that renders it to a link is unaware of any reference data provided elsewhere in that text file or otherwise.

I was not talking about the expansion of something like `{{mr:123456}}`, but instead about this list of references in a "References" page.  The contents of that page is constructed from the `name:` fields from the metadata.

File: spaces/S000147/properties/P000097.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 space: S000147
-property: P000057
-value: false
+property: P000097
```
Comment (StevenClontz) [inline spaces/S000147/properties/P000097.md:3]: We could add https://topology.pi-base.org/properties/P000065 and https://topology.pi-base.org/properties/P000114 (the space assumes existence of CH).

Comment (prabau) [reply spaces/S000147/properties/P000097.md:3]: I guess so, and also for S148 (the other Luzin space).  I'll add it.

Comment (prabau) [reply spaces/S000147/properties/P000097.md:3]: Question: For spaces whole definition relies on extra set-theoretic assumptions, how about adding them in parentheses at the beginning of the space name?  So in this case: `(CH) A Luzin subset of the reals (Scheepers 1999)` for example.

File: theorems/T000232.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+uid: T000232
+if:
+  and:
+  - P000046: true
+  - P000097: true
+then:
+  P000050: true
+refs:
+  - mathse: 3824535
```
Comment (StevenClontz) [inline theorems/T000232.md:10]: I think https://math.stackexchange.com/questions/3824078/a-subspace-of-mathbbr-is-zero-dimensional-iff-it-is-totally-disconnected/3824535#comment7886688_3824078 suggests "embeddable in R" can be generalized to a "GO-space" (subspace of any LOTS). perhaps a future contribution.

Comment (prabau) [reply theorems/T000232.md:10]: Absolutely.  That was my plan after this: add GO-space as a property. Many theorems we already have for LOTS can be generalized for those.

**Review by StevenClontz — APPROVED** [review]

File: spaces/S000147/properties/P000097.md (hunk 2 above)
Comment (StevenClontz) [reply spaces/S000147/properties/P000097.md:3]: I like that

**Comment by StevenClontz** [comment]
> > The issue is that when you write `{{mr:123456}}`, the component that renders it to a link is unaware of any reference data provided elsewhere in that text file or otherwise.
> 
> I was not talking about the expansion of something like `{{mr:123456}}`, but instead about the list of references in a "References" page. The contents of that list is constructed from the `name:` fields in the `refs:` section from the metadata.

Oh gotcha. Open an issue in /web and assign me and I can maybe take a crack at this.

**Comment by StevenClontz** [comment]
Approved - make any discussed tweaks then merge when you're ready.

**Comment by prabau** [comment]
@StevenClontz Why did you approve?  I was still going to add the few things you suggested.

**Comment by prabau** [comment]
Hmm, thanks for approving.  But I think it would have been better to let Chris do it, as he was planning to do it.  Also, when there is still work going on, seems better for that to be complete.

**Comment by StevenClontz** [comment]
Assigning to @ccaruvana

**Comment by ccaruvana** [comment]
Haha I didn't mean to claim ownership over approving, but thanks for keeping me in on the loop. Based on the conversation, this PR seems ready to go and the GO space will be added later (go GO).

**Review by ccaruvana — APPROVED** [review]

**Comment by Jianing-Song** [comment]
As for a converse of LOTS + second countable ==> embeddable in $\mathbb{R}$, we need a space embeddable in $\mathbb{R}$ that is not a LOTS.

I propose the space $X$ that is the disjoint union of $\mathbb{R}$ and an extra point $\infty$. It is clearly embeddable in $\mathbb{R}$ as $(0,1)\cup\\{2\\}$. To show that it is not a LOTS, note that in the opposite situation $\infty$ must be either the largest or the smallest element of $X$, otherwise $(\left,\infty)$ and $(\infty,\right)$ is an open partition of $X\setminus\]{\infty\\}$ which is connected, impossible.

Suppose WLOG that $\infty$ is the largest element, then $\{\infty\}$ is open implies that $\mathbb{R}$ contains a largest element $\{a\}$. But then $\mathbb{R}\setminus\\{a\\}$ is still connected in the new order topology (removing the largest element of a connected LOTS leaves a connected space, since for LOTS connectedness <=> dense order + every $[a,b]$ is compact; also we can see that the least-upper-bound property is preserved because a set with LUB $\\{a\\}$ is now unbounded in $\mathbb{R}\setminus\\{a\\}$), and removing any point from $\mathbb{R}$ results in a disconnected space in the usual topology.

**Comment by prabau** [comment]
@Jianing-Song Good example to have, with great justification here.

One minor thing.  It's true that for a connected LOTS with a maximum element, removing that maximum still gives a connected space.  But connected LOTS is not exactly that same as order-dense + order-complete.  Order-complete means every subset has a supremum.  But we need instead: every subset with an upper bound has a least upper bound.  It's what the wikipedia article calls "bounded complete" (not sure how common that terminology is though).

**Comment by Jianing-Song** [comment]
> @Jianing-Song Good example to have, with great justification here.
> 
> One minor thing. It's true that for a connected LOTS with a maximum element, removing that maximum still gives a connected space. But connected LOTS is not exactly that same as order-dense + order-complete. Order-complete means every subset has a supremum. But we need instead: every subset with an upper bound has a least upper bound. It's what the wikipedia article calls "bounded complete" (not sure how common that terminology is though).

OK, but then being order-complete is just the same thing as being compact. Strange enough!

**Comment by prabau** [comment]
Yes, order-complete is the same as being compact.

**Comment by prabau** [comment]
I missed one detail.  The "bounded complete" property, more commonly called "least upper bound property" it seems, should be:
Every **nonempty** subset with an upper bound has a least upper bound.

(Seems wikipedia is missing this nonempty qualifications in a few places)

**Pull request merged**